packages/@vue/cli/lib/Generator.js

```diff
@@ -97,17 +97,31 @@ module.exports = class Generator {
     const rootOptions = cliService
       ? cliService.options
       : inferRootOptions(pkg)
-    // apply generators from plugins
-    plugins.forEach(({ id, apply, options }) => {
-      const api = new GeneratorAPI(id, this, options, rootOptions)
-      apply(api, options, rootOptions, invoking)
+
+    this.rootOptions = rootOptions
+  }
+
+  initPlugins () {
+    const { rootOptions, invoking } = this
+    return new Promise((resolve, reject) => {
```

Comment:
Maybe use `async` with a `for` loop?

Reply:
Can we not do something like:
```suggestion
    return Promise.all(this.plugins.map(({ id, apply, options }) => {
        const api = new GeneratorAPI(id, this, options, rootOptions)
        return apply(api, options, rootOptions, invoking)
      }))
```